---
title: "Step-by-step Chinook"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  github_document:
    toc: true
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, echo = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "chinook-step-by-step-figs/"
)
```

```{r libs}
library(tidyverse)
```
## Introduction

This is going to follow along what we did with mykiss (albeit more tersely, as the
explanations were all done there.)

The genome bowtie data base is already built so we start with mapping.

I note that I am going to do this all within the Mykss directory, which is a little
nonstandard, but keeps all the fish stuff in one place.  

## Mapping

I moved the chinook meta data to `./chinook-scripts/other_inputs/chinook_meta.xls`. Then I copied 
the entire contents out of the "Table S7 -- Chinook Rad Sequence" from that file and did this:
```{sh, eval=FALSE}
2017-01-17 05:26 /chinook_scripts/--% (master) pwd
/Users/eriq/Documents/git-repos/genoscape-bioinformatics/chinook-scripts
2017-01-17 05:26 /chinook_scripts/--% (master) pbpaste | tr '\r' '\n' | awk 'BEGIN {print "0 ID PU SM PL LB"} NR > 1 {print ++n,$1, $3"."$4, $NF, "ILLUMINA", $4}' > chinook_ids.txt
```
So, now we have a file called `./chinook-scripts/chinook_ids.txt` that looks like this:
```
0 ID PU SM PL LB
1 SOMM042_Index04_AAACGGTGCAGG SOMM042.Index04 DNAA025_A01 ILLUMINA Index04
2 SOMM042_Index04_AACGTTTGCAGG SOMM042.Index04 DNAA025_A02 ILLUMINA Index04
3 SOMM042_Index04_AACTGATGCAGG SOMM042.Index04 DNAA025_A03 ILLUMINA Index04
4 SOMM042_Index04_AAGACGTGCAGG SOMM042.Index04 DNAA025_A04 ILLUMINA Index04
5 SOMM042_Index04_AAGCTATGCAGG SOMM042.Index04 DNAA025_A05 ILLUMINA Index04
6 SOMM042_Index04_AATATCTGCAGG SOMM042.Index04 DNAA025_A06 ILLUMINA Index04
7 SOMM042_Index04_AATGAGTGCAGG SOMM042.Index04 DNAA025_A07 ILLUMINA Index04
8 SOMM042_Index04_ACAAGATGCAGG SOMM042.Index04 DNAA025_A08 ILLUMINA Index04
9 SOMM042_Index04_ACAGCGTGCAGG SOMM042.Index04 DNAA025_A09 ILLUMINA Index04
...
```
which is totally ready for mapping with bowtie in a job array. I make the script,
`/chinook_scripts/02-chinook-bowtie-map-job-array.sh` which is like the mykiss version but we set the 
number of jobs to 283 (the number of chinook samples) and change the path to the 
IDFILE.

It should be noted that I might want to use stampy instead of bowtie for aligning to a different
species, but I have the bowtie script written already so I just want to see how it goes.
```{sh, eval=FALSE}
[kruegg@login3 Chinook_all_preps]$ pwd
/u/home/k/kruegg/nobackup-klohmuel/Mykiss/Chinook_all_preps
[kruegg@login3 Chinook_all_preps]$ qsub ~/genoscape-bioinformatics/chinook-scripts/02-chinook-bowtie-map-job-array.sh 
JSV: PE=shared
Your job-array 1509154.1-283:1 ("radMap") has been submitted
```
Unfortunately, the first time I ran it I had an error in the path to the chinook scripts 
and it didn't find the IDFILE and it ran through all the jobs really fast with no errors so
we got this:
```
!!! Please note !!!
Your account is being throttled because it has ran too many short jobs
in the last few hours. Run this command to see the start/end times of
your recent jobs: qacct -o kruegg -j -d 1

See http://goo.gl/dQXZHe for a technique to avoid this in the future.

Once your account stops running so many short jobs, the system will
automatically put it back to normal state (within a few hours).

Please understand that we are throttling short jobs to prevent the
scheduler (software) from crashing.
Contact http://support.idre.ucla.edu/helpdesk if you have questions.
```
So, good reminder that I should always test a job array by running it up to 2 or 3 first...

Eventually that started up and ran pretty quickly, except for #5 which was
`SOMM042_Index04_AAGCTATGCAGG`, a fish with a boatload of sequence. So, everything
else has finished, but it is still crunching away...then finally finished.

## Dup-filtering

Doing this as a job array, but doing more than 10 on each node so that the jobs
are sure to not be too short.
```{sh, eval=FALSE}
[kruegg@login2 bam]$ pwd
/u/home/k/kruegg/nobackup-klohmuel/Mykiss/Chinook_all_preps/alignments/omyV6/bam
[kruegg@login2 bam]$ ls -l  *.bam | awk 'NR % 20 == 1 {printf("\n%d ",++n)} {printf(" %s", $NF);} END {printf("\n");}' > bamlist.txt 

[kruegg@login2 bam]$ head -n 3 bamlist.txt 

1  SOMM042_Index04_AAACGGTGCAGG.bam SOMM042_Index04_AACGTTTGCAGG.bam SOMM042_Index04_AACTGATGCAGG.bam SOMM042_Index04_AAGACGTGCAGG.bam SOMM042_Index04_AAGCTATGCAGG.bam SOMM042_Index04_AATATCTGCAGG.bam SOMM042_Index04_AATGAGTGCAGG.bam SOMM042_Index04_ACAAGATGCAGG.bam SOMM042_Index04_ACAGCGTGCAGG.bam SOMM042_Index04_ACATACTGCAGG.bam SOMM042_Index04_ACCATGTGCAGG.bam SOMM042_Index04_ACCCCCTGCAGG.bam SOMM042_Index04_ACTCTTTGCAGG.bam SOMM042_Index04_ACTGGCTGCAGG.bam SOMM042_Index04_AGCCATTGCAGG.bam SOMM042_Index04_AGCGCATGCAGG.bam SOMM042_Index04_AGGGTCTGCAGG.bam SOMM042_Index04_AGGTGTTGCAGG.bam SOMM042_Index04_AGTAGGTGCAGG.bam SOMM042_Index04_AGTTAATGCAGG.bam
2  SOMM042_Index04_ATAGTATGCAGG.bam SOMM042_Index04_ATCAAATGCAGG.bam SOMM042_Index04_ATGCACTGCAGG.bam SOMM042_Index04_ATGTTGTGCAGG.bam SOMM042_Index04_ATTCCGTGCAGG.bam SOMM042_Index04_CAAAAATGCAGG.bam SOMM042_Index04_CAATCGTGCAGG.bam SOMM042_Index04_CACCTCTGCAGG.bam SOMM042_Index04_CAGGCATGCAGG.bam SOMM042_Index04_CATACTTGCAGG.bam SOMM042_Index04_CCATTTTGCAGG.bam SOMM042_Index04_CCCGGTTGCAGG.bam SOMM042_Index04_CCCTAATGCAGG.bam SOMM042_Index04_CCGAGGTGCAGG.bam SOMM042_Index04_CCGCATTGCAGG.bam SOMM042_Index04_CCTAACTGCAGG.bam SOMM042_Index04_CGAGGCTGCAGG.bam SOMM042_Index04_CGCAGATGCAGG.bam SOMM042_Index04_CGCGTGTGCAGG.bam SOMM042_Index04_CGGTCCTGCAGG.bam

```
and there are 15 lines in there.

Then launch that dude.
```{sh, eval=FALSE}
[kruegg@login2 bam]$ pwd
/u/home/k/kruegg/nobackup-klohmuel/Mykiss/Chinook_all_preps/alignments/omyV6/bam
[kruegg@login2 bam]$ qsub ~/genoscape-bioinformatics/chinook-scripts/03-chinook-rmdup-alignments-job-array.sh 
JSV: PE=shared
Your job-array 1510819.1-15:1 ("rmdup") has been submitted
[kruegg@login2 bam]$ date
Tue Jan 17 10:47:05 PST 2017
```
That takes about 20 minutes.

## Merging Bams

One script:
```{sh, eval=FALSE}
[kruegg@login2 omyV6]$ pwd
/u/home/k/kruegg/nobackup-klohmuel/Mykiss/Chinook_all_preps/alignments/omyV6
[kruegg@login2 omyV6]$ qsub ~/genoscape-bioinformatics/chinook-scripts/04-chinook-merge-bams.sh 
JSV: PE=shared
Your job 1511200 ("merge-bams") has been submitted
[kruegg@login2 omyV6]$ date
Tue Jan 17 11:14:19 PST 2017
```
That took almost 2 hours.

## Indexing the Genome Fasta and creating a dictionary

This was already done for mykiss.  So, no need to do it over here.

## Calling SNPs with GATK

Make a SNPs directory and get prepared.
```{sh, eval=FALSE}
[kruegg@login2 Chinook_all_preps]$ mkdir SNPs
[kruegg@login2 Chinook_all_preps]$ pwd
/u/home/k/kruegg/nobackup-klohmuel/Mykiss/Chinook_all_preps

[kruegg@login2 Chinook_all_preps]$ cd SNPs/
[kruegg@login2 SNPs]$ module load samtools
[kruegg@login2 SNPs]$ pwd
/u/home/k/kruegg/nobackup-klohmuel/Mykiss/Chinook_all_preps/SNPs

[kruegg@login2 SNPs]$ BAM=../alignments/omyV6/MergedBams/omyV6-merged.bam

[kruegg@login2 SNPs]$ samtools view -H $BAM | awk 'BEGIN {OFS="\t"} /@SQ/ {print ++n, $2}' | sed 's/SN://g' > chromo_list.txt 
[kruegg@login2 SNPs]$ head chromo_list.txt 
1	omy01
2	omy02
3	omy03
4	omy04
5	omy05
6	omy06
7	omy07
8	omy08
9	omy09
10	omy10

```

