---
title: "Step-by-step Chinook"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  github_document:
    toc: true
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, echo = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "chinook-step-by-step-figs/"
)
```

```{r libs}
library(tidyverse)
```
## Introduction

This is going to follow along what we did with mykiss (albeit more tersely, as the
explanations were all done there.)

The genome bowtie data base is already built so we start with mapping.

I note that I am going to do this all within the Mykss directory, which is a little
nonstandard, but keeps all the fish stuff in one place.  

## Mapping

I moved the chinook meta data to `./chinook-scripts/other_inputs/chinook_meta.xls`. Then I copied 
the entire contents out of the "Table S7 -- Chinook Rad Sequence" from that file and did this:
```{sh, eval=FALSE}
2017-01-17 05:26 /chinook_scripts/--% (master) pwd
/Users/eriq/Documents/git-repos/genoscape-bioinformatics/chinook-scripts
2017-01-17 05:26 /chinook_scripts/--% (master) pbpaste | tr '\r' '\n' | awk 'BEGIN {print "0 ID PU SM PL LB"} NR > 1 {print ++n,$1, $3"."$4, $NF, "ILLUMINA", $4}' > chinook_ids.txt
```
So, now we have a file called `./chinook-scripts/chinook_ids.txt` that looks like this:
```
0 ID PU SM PL LB
1 SOMM042_Index04_AAACGGTGCAGG SOMM042.Index04 DNAA025_A01 ILLUMINA Index04
2 SOMM042_Index04_AACGTTTGCAGG SOMM042.Index04 DNAA025_A02 ILLUMINA Index04
3 SOMM042_Index04_AACTGATGCAGG SOMM042.Index04 DNAA025_A03 ILLUMINA Index04
4 SOMM042_Index04_AAGACGTGCAGG SOMM042.Index04 DNAA025_A04 ILLUMINA Index04
5 SOMM042_Index04_AAGCTATGCAGG SOMM042.Index04 DNAA025_A05 ILLUMINA Index04
6 SOMM042_Index04_AATATCTGCAGG SOMM042.Index04 DNAA025_A06 ILLUMINA Index04
7 SOMM042_Index04_AATGAGTGCAGG SOMM042.Index04 DNAA025_A07 ILLUMINA Index04
8 SOMM042_Index04_ACAAGATGCAGG SOMM042.Index04 DNAA025_A08 ILLUMINA Index04
9 SOMM042_Index04_ACAGCGTGCAGG SOMM042.Index04 DNAA025_A09 ILLUMINA Index04
...
```
which is totally ready for mapping with bowtie in a job array. I make the script,
`/chinook_scripts/02-chinook-bowtie-map-job-array.sh` which is like the mykiss version but we set the 
number of jobs to 283 (the number of chinook samples) and change the path to the 
IDFILE.

It should be noted that I might want to use stampy instead of bowtie for aligning to a different
species, but I have the bowtie script written already so I just want to see how it goes.
```{sh, eval=FALSE}
[kruegg@login3 Chinook_all_preps]$ pwd
/u/home/k/kruegg/nobackup-klohmuel/Mykiss/Chinook_all_preps
[kruegg@login3 Chinook_all_preps]$ qsub ~/genoscape-bioinformatics/chinook-scripts/02-chinook-bowtie-map-job-array.sh 
JSV: PE=shared
Your job-array 1509154.1-283:1 ("radMap") has been submitted
```
Unfortunately, the first time I ran it I had an error in the path to the chinook scripts 
and it didn't find the IDFILE and it ran through all the jobs really fast with no errors so
we got this:
```
!!! Please note !!!
Your account is being throttled because it has ran too many short jobs
in the last few hours. Run this command to see the start/end times of
your recent jobs: qacct -o kruegg -j -d 1

See http://goo.gl/dQXZHe for a technique to avoid this in the future.

Once your account stops running so many short jobs, the system will
automatically put it back to normal state (within a few hours).

Please understand that we are throttling short jobs to prevent the
scheduler (software) from crashing.
Contact http://support.idre.ucla.edu/helpdesk if you have questions.
```
So, good reminder that I should always test a job array by running it up to 2 or 3 first...

Eventually that started up and ran pretty quickly, except for #5 which was
`SOMM042_Index04_AAGCTATGCAGG`, a fish with a boatload of sequence. So, everything
else has finished, but it is still crunching away...then finally finished.

## Dup-filtering

Doing this as a job array, but doing more than 10 on each node so that the jobs
are sure to not be too short.
```{sh, eval=FALSE}
[kruegg@login2 bam]$ pwd
/u/home/k/kruegg/nobackup-klohmuel/Mykiss/Chinook_all_preps/alignments/omyV6/bam
[kruegg@login2 bam]$ ls -l  *.bam | awk 'NR % 20 == 1 {printf("\n%d ",++n)} {printf(" %s", $NF);} END {printf("\n");}' > bamlist.txt 

[kruegg@login2 bam]$ head -n 3 bamlist.txt 

1  SOMM042_Index04_AAACGGTGCAGG.bam SOMM042_Index04_AACGTTTGCAGG.bam SOMM042_Index04_AACTGATGCAGG.bam SOMM042_Index04_AAGACGTGCAGG.bam SOMM042_Index04_AAGCTATGCAGG.bam SOMM042_Index04_AATATCTGCAGG.bam SOMM042_Index04_AATGAGTGCAGG.bam SOMM042_Index04_ACAAGATGCAGG.bam SOMM042_Index04_ACAGCGTGCAGG.bam SOMM042_Index04_ACATACTGCAGG.bam SOMM042_Index04_ACCATGTGCAGG.bam SOMM042_Index04_ACCCCCTGCAGG.bam SOMM042_Index04_ACTCTTTGCAGG.bam SOMM042_Index04_ACTGGCTGCAGG.bam SOMM042_Index04_AGCCATTGCAGG.bam SOMM042_Index04_AGCGCATGCAGG.bam SOMM042_Index04_AGGGTCTGCAGG.bam SOMM042_Index04_AGGTGTTGCAGG.bam SOMM042_Index04_AGTAGGTGCAGG.bam SOMM042_Index04_AGTTAATGCAGG.bam
2  SOMM042_Index04_ATAGTATGCAGG.bam SOMM042_Index04_ATCAAATGCAGG.bam SOMM042_Index04_ATGCACTGCAGG.bam SOMM042_Index04_ATGTTGTGCAGG.bam SOMM042_Index04_ATTCCGTGCAGG.bam SOMM042_Index04_CAAAAATGCAGG.bam SOMM042_Index04_CAATCGTGCAGG.bam SOMM042_Index04_CACCTCTGCAGG.bam SOMM042_Index04_CAGGCATGCAGG.bam SOMM042_Index04_CATACTTGCAGG.bam SOMM042_Index04_CCATTTTGCAGG.bam SOMM042_Index04_CCCGGTTGCAGG.bam SOMM042_Index04_CCCTAATGCAGG.bam SOMM042_Index04_CCGAGGTGCAGG.bam SOMM042_Index04_CCGCATTGCAGG.bam SOMM042_Index04_CCTAACTGCAGG.bam SOMM042_Index04_CGAGGCTGCAGG.bam SOMM042_Index04_CGCAGATGCAGG.bam SOMM042_Index04_CGCGTGTGCAGG.bam SOMM042_Index04_CGGTCCTGCAGG.bam

```
and there are 15 lines in there.

Then launch that dude.
```{sh, eval=FALSE}
[kruegg@login2 bam]$ pwd
/u/home/k/kruegg/nobackup-klohmuel/Mykiss/Chinook_all_preps/alignments/omyV6/bam
[kruegg@login2 bam]$ qsub ~/genoscape-bioinformatics/chinook-scripts/03-chinook-rmdup-alignments-job-array.sh 
JSV: PE=shared
Your job-array 1510819.1-15:1 ("rmdup") has been submitted
[kruegg@login2 bam]$ date
Tue Jan 17 10:47:05 PST 2017
```
That takes about 20 minutes.

## Merging Bams

One script:
```{sh, eval=FALSE}
[kruegg@login2 omyV6]$ pwd
/u/home/k/kruegg/nobackup-klohmuel/Mykiss/Chinook_all_preps/alignments/omyV6
[kruegg@login2 omyV6]$ qsub ~/genoscape-bioinformatics/chinook-scripts/04-chinook-merge-bams.sh 
JSV: PE=shared
Your job 1511200 ("merge-bams") has been submitted
[kruegg@login2 omyV6]$ date
Tue Jan 17 11:14:19 PST 2017
```
That took almost 2 hours.

## Indexing the Genome Fasta and creating a dictionary

This was already done for mykiss.  So, no need to do it over here.

## Calling SNPs with GATK

Make a SNPs directory and get prepared.
```{sh, eval=FALSE}
[kruegg@login2 Chinook_all_preps]$ mkdir SNPs
[kruegg@login2 Chinook_all_preps]$ pwd
/u/home/k/kruegg/nobackup-klohmuel/Mykiss/Chinook_all_preps

[kruegg@login2 Chinook_all_preps]$ cd SNPs/
[kruegg@login2 SNPs]$ module load samtools
[kruegg@login2 SNPs]$ pwd
/u/home/k/kruegg/nobackup-klohmuel/Mykiss/Chinook_all_preps/SNPs

[kruegg@login2 SNPs]$ BAM=../alignments/omyV6/MergedBams/omyV6-merged.bam

[kruegg@login2 SNPs]$ samtools view -H $BAM | awk 'BEGIN {OFS="\t"} /@SQ/ {print ++n, $2}' | sed 's/SN://g' > chromo_list.txt 
[kruegg@login2 SNPs]$ head chromo_list.txt 
1	omy01
2	omy02
3	omy03
4	omy04
5	omy05
6	omy06
7	omy07
8	omy08
9	omy09
10	omy10

```
Then launch that dude:
```{sh, eval=FALSE}
[kruegg@login2 SNPs]$ pwd
/u/home/k/kruegg/nobackup-klohmuel/Mykiss/Chinook_all_preps/SNPs
[kruegg@login2 SNPs]$ qsub ~/genoscape-bioinformatics/chinook-scripts/05-chinook-call-snps-array.sh 
JSV: PE=shared
Your job-array 1512271.1-29:1 ("snp-array") has been submitted
[kruegg@login2 SNPs]$ date
Tue Jan 17 13:09:13 PST 2017
```
It looks like it is only going to launch four nodes at once.  Probably much higher load
on hoffman than when I did the mykiss.  But then it launched a bunch and finished quite
quickly.

After it ran, I moved all the output (and input) to `/u/home/k/kruegg/nobackup-klohmuel/Mykiss/Chinook_all_preps/SNPs/chromo_pieces`.
```{sh, eval=FALSE}
[kruegg@n2192 SNPs]$ pwd 
/u/home/k/kruegg/nobackup-klohmuel/Mykiss/Chinook_all_preps/SNPs
[kruegg@n2192 SNPs]$ mkdir chromo_pieces
[kruegg@n2192 SNPs]$ mv * chromo_pieces/
mv: cannot move `chromo_pieces' to a subdirectory of itself, `chromo_pieces/chromo_pieces'
[kruegg@n2192 SNPs]$ ls
chromo_pieces
```
And then, on a compute node, I merged everything into a single merged vcf:
```{sh, eval=FALSE}
[kruegg@n2192 chromo_pieces]$ pwd
/u/home/k/kruegg/nobackup-klohmuel/Mykiss/Chinook_all_preps/SNPs/chromo_pieces
[kruegg@n2192 chromo_pieces]$ PICARD_JAR=/u/nobackup/klohmuel/kruegg/bin/picard.jar
[kruegg@n2192 chromo_pieces]$ INPUTS=$(ls -l 0*.vcf | awk '{printf("I=%s ", $NF)}') 
[kruegg@n2192 chromo_pieces]$ echo $INPUTS
I=001.vcf I=002.vcf I=003.vcf I=004.vcf I=005.vcf I=006.vcf I=007.vcf I=008.vcf I=009.vcf I=010.vcf I=011.vcf I=012.vcf I=013.vcf I=014.vcf I=015.vcf I=016.vcf I=017.vcf I=018.vcf I=019.vcf I=020.vcf I=021.vcf I=022.vcf I=023.vcf I=024.vcf I=025.vcf I=026.vcf I=027.vcf I=028.vcf I=029.vcf
[kruegg@n2192 chromo_pieces]$ module load java
[kruegg@n2192 chromo_pieces]$ java -jar $PICARD_JAR SortVcf $INPUTS O=chinook-full-omyV6.vcf
[Tue Jan 17 20:53:08 PST 2017] picard.vcf.SortVcf INPUT=[001.vcf, 002.vcf, 003.vcf, 004.vcf, 005.vcf, 006.vcf, 007.vcf, 008.vcf, 009.vcf, 010.vcf, 011.vcf, 012.vcf, 013.vcf, 014.vcf, 015.vcf, 016.vcf, 017.vcf, 018.vcf, 019.vcf, 020.vcf, 021.vcf, 022.vcf, 023.vcf, 024.vcf, 025.vcf, 026.vcf, 027.vcf, 028.vcf, 029.vcf] OUTPUT=chinook-full-omyV6.vcf    VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=true CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json
[Tue Jan 17 20:53:08 PST 2017] Executing as kruegg@n2192 on Linux 2.6.32-642.11.1.el6.x86_64 amd64; Java HotSpot(TM) 64-Bit Server VM 1.8.0_111-b14; Picard version: 2.7.1-SNAPSHOT
INFO	2017-01-17 20:53:10	SortVcf	Reading entries from input file 1
INFO	2017-01-17 20:53:11	SortVcf	read        25,000 records.  Elapsed time: 00:00:01s.  Time for last 25,000:    1s.  Last read position: omy01:57,909,909
INFO	2017-01-17 20:53:12	SortVcf	Reading entries from input file 2
INFO	2017-01-17 20:53:12	SortVcf	read        50,000 records.  Elapsed time: 00:00:02s.  Time for last 25,000:    1s.  Last read position: omy02:35,641,871
INFO	2017-01-17 20:53:15	SortVcf	Reading entries from input file 3
...
[kruegg@n2192 chromo_pieces]$ mv chinook-full-omyV6.vcf* ../
```
Now, the file is in `/u/nobackup/klohmuel/kruegg/Mykiss/Chinook_all_preps/SNPs/chinook-full-omyV6.vcf`.

### Light SNP Filtering

As before, I do a very light filtering:
```{sh, eval=FALSE}
[kruegg@n2196 SNPs]$ module load vcftools
[kruegg@n2196 SNPs]$ pwd
/u/home/k/kruegg/nobackup-klohmuel/Mykiss/Chinook_all_preps/SNPs
[kruegg@n2196 SNPs]$ vcftools --vcf chinook-full-omyV6.vcf --out chinook-light  --remove-indels --min-alleles 2 --max-alleles 2 --maf 0.05  --max-missing 0.1 --recode

VCFtools - 0.1.14
(C) Adam Auton and Anthony Marcketta 2009

Parameters as interpreted:
	--vcf chinook-full-omyV6.vcf
	--maf 0.05
	--max-alleles 2
	--min-alleles 2
	--max-missing 0.1
	--out chinook-light
	--recode
	--remove-indels

After filtering, kept 250 out of 250 Individuals
Outputting VCF file...
After filtering, kept 56220 out of a possible 758590 Sites
Run Time = 97.00 seconds
[kruegg@n2196 SNPs]$ gzip chinook-light.recode.vcf 
[kruegg@n2196 SNPs]$ du -h chinook-light.recode.vcf.gz 
20M	chinook-light.recode.vcf.gz
```
Then I bring that to my laptop:
`/Users/eriq/Documents/git-repos/genoscape-bioinformatics/chinook-scripts/plink/chinook-light.recode.vcf.gz`

## Association Test Stuff

For now I am not going to worry about accounting for population structure. I am just
going to do what Mike and company did for the first submission, which is throw everyone in
as either Mature or Premature, regardless of population.  

### Further filtering

But we will do some continued filtering:
drop indivs with more than 70% missing data and SNPs missing in more than 50% of the indivs:
```{sh, eval=FALSE}
2017-01-18 06:00 /plink/--% (master) pwd
/Users/eriq/Documents/git-repos/genoscape-bioinformatics/chinook-scripts/plink
2017-01-18 06:00 /plink/--% (master) plink --vcf chinook-light.recode.vcf.gz  --aec --mind 0.7 --geno 0.5   --make-bed --out chinook33K_184 --double-id
PLINK v1.90b3.42 64-bit (20 Sep 2016)      https://www.cog-genomics.org/plink2
(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to chinook33K_184.log.
Options in effect:
  --allow-extra-chr
  --double-id
  --geno 0.5
  --make-bed
  --mind 0.7
  --out chinook33K_184
  --vcf chinook-light.recode.vcf.gz

4096 MB RAM detected; reserving 2048 MB for main workspace.
--vcf: chinook33K_184-temporary.bed + chinook33K_184-temporary.bim +
chinook33K_184-temporary.fam written.
56220 variants loaded from .bim file.
250 people (0 males, 0 females, 250 ambiguous) loaded from .fam.
Ambiguous sex IDs written to chinook33K_184.nosex .
66 people removed due to missing genotype data (--mind).
IDs written to chinook33K_184.irem .
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 184 founders and 0 nonfounders present.
Calculating allele frequencies... done.
Total genotyping rate in remaining samples is 0.558651.
22593 variants removed due to missing genotype data (--geno).
33627 variants and 184 people pass filters and QC.
Note: No phenotypes present.
--make-bed to chinook33K_184.bed + chinook33K_184.bim + chinook33K_184.fam ...
done.
```
That leaves us with only 33,627 SNPs in 184 fish  But with luck we will still have 
the SNPs that Mike et al. found.

### Defining cases
So, first we are going to need
to figure out who is whom from the meta data:
```{r figure-meta}
# read in the meta data
chinook_meta <- readxl::read_excel(path = "../../chinook-scripts/other_inputs/chinook_meta.xls", sheet = 1)


# get a list of the individuals we have retained from the genomic data
retained <- read_delim("../../chinook-scripts/plink/chinook33K_184.fam", col_names = FALSE, delim = " ") %>%
  select(X1) %>%
  unlist() %>% unname()

# now pick out only those in the retained file
chinook_retained <- chinook_meta %>%
  filter(`Sample DNA ID` %in% retained)

# now make a file of the cases
chinook_retained %>%
  filter(`Migration Category` == "Premature") %>%
  mutate(a1 = `Sample DNA ID`, a2 = `Sample DNA ID`) %>%
  select(a1, a2) %>% 
  write.table(., quote = F, sep = " ", row.names = F, col.names = F, file = "../../chinook-scripts/plink/chinook_cases.txt") 

```

### Association Test
And now let's do a quick association test:
```{sh, eval=FALSE}
2017-01-18 06:16 /plink/--% (master) pwd
/Users/eriq/Documents/git-repos/genoscape-bioinformatics/chinook-scripts/plink
2017-01-18 06:16 /plink/--% (master) plink -bfile chinook33K_184  --aec --make-pheno chinook_cases.txt '*'  --assoc fisher --out chinook_ass --allow-no-sex
PLINK v1.90b3.42 64-bit (20 Sep 2016)      https://www.cog-genomics.org/plink2
(C) 2005-2016 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to chinook_ass.log.
Options in effect:
  --allow-extra-chr
  --allow-no-sex
  --assoc fisher
  --bfile chinook33K_184
  --make-pheno chinook_cases.txt *
  --out chinook_ass

4096 MB RAM detected; reserving 2048 MB for main workspace.
33627 variants loaded from .bim file.
184 people (0 males, 0 females, 184 ambiguous) loaded from .fam.
Ambiguous sex IDs written to chinook_ass.nosex .
--make-pheno: 184 phenotype values set.
Using 1 thread (no multithreaded calculations invoked).
Before main variant filters, 184 founders and 0 nonfounders present.
Calculating allele frequencies... done.
Total genotyping rate is 0.721972.
33627 variants and 184 people pass filters and QC.
Among remaining phenotypes, 84 are cases and 100 are controls.
Writing C/C --assoc report to chinook_ass.assoc.fisher ...
done.
```

### Looking at the results

```{r ass-looker}
chin_ass <- read.table("../../chinook-scripts/plink/chinook_ass.assoc.fisher", header=T, stringsAsFactors = F)

arrange(chin_ass, P) %>% head(n=20)
```

OK, that looks like 2 SNPs that we could be really excited about on omy28.  Though I am not sure how this 
corresponds with what Prince et al found.  Let's grab the top four in order along omy28
```{r get-four}
top4 <- chin_ass %>%
  top_n(n = 4, wt = -P) %>%
  arrange(BP)

top4
```

And let us compare that to what Mike and company found when they were ignoring population structure:
```{r mike-four}
mike4 <- readxl::read_excel("../../chinook-scripts/other_inputs/chinook-meta-submission-1-biorxiv.xls", sheet = 3)
as.data.frame(mike4)
```

Wow! It doesn't look like those correspond well at all.  Let's look at the diffs:
```{r check-diffs}
tbp <- top4$BP
mbp <- mike4$`Scaffold79929e Position`
names(tbp) <- tbp
names(mbp) <- mbp

diffmat <- outer(tbp, mbp, "-")
diffmat
```

So, nothing is really lining up there.  But at least we have some candidates if we want to try to 
find these and do some resequencing.

### Grabbing some flanking sequences

Let's just go ahead and grab 999 bp to the left and 1000 bp to the right of each of those, just
to have it.
```{r grab-string}
top4 %>%
  mutate(chrstr = paste(CHR, ":", BP - 999, "-", BP + 1000, sep = "")) %>%
  select(chrstr) %>%
  unlist() %>%
  cat(sep = " ")
```
And we take that and grab it out of the fasta:
```{sh, eval=FALSE}
[kruegg@n2196 Genome]$ pwd
/u/home/k/kruegg/nobackup-klohmuel/Mykiss/Genome
[kruegg@n2196 Genome]$ module load samtools
[kruegg@n2196 Genome]$ samtools faidx omyV6Chr.fasta  omy28:11588348-11590347 omy28:11608637-11610636 omy28:11856799-11858798 omy28:12666405-12668404
>omy28:11588348-11590347
TATAGACACCCTAACAGTGTGTGCACATGCAGGGAAATACAGTAGACACACTGTACACAC
ACAGTACACACACTCTGCAAAAGCACACACGGTGAACTCTTCTTGTCTTTTGTGTGCCAC
CCAGACCCCCAGAACGTACTGCGCTGCCAACACAGAGGACTTGGAGGCTGTCATCGAACA
CATCCAGAGGACAAATGAGGACGCCCCTCTCATGGCCGCTGGGGTCTCCATGGGCGGGTA
GGTGCCTCTTCATGCCTTCTCACTTCAATGGACAGTGACACGATACACGGGGCGAAAATC
CCAAGCACGTGCATTACTCAGACTTATGCATCCATCGTGTATTATTGTAGCTTATTGACG
TATTACTTCACTGAACTGTGGGAGTCAGTCGGAGGCGAGTACGGCTTATGTAGAGCCTCG
AGAGCGCACTTGGTCATCCTTAGGTTTTTAATGATATTCAGAAAAGGTCAACGCACACCA
GTAGAAAGGATGTCATGGGTTGGTTATGTATTTTTGACATTTTGCAGCCTCATGCTTTCC
CCCTGTCAGAAAACAACACTTTTCAGTTGATTCAGAGAGCTCAGCCTCTCAGTAGTGCTT
AGACCATGTCAAAAGAGGTATTCAAGCTATAATCTCCTAGCTAAGTGCAGATTGATGAAA
AATGTAACATTTCACAATCATCCTCTGCTCTCCCCCCTCCCTTCTCTCCACGACCCTCTT
CCCCCTCTTTTGAACCCCAGGATGATGCTGGCAAACTACCTGGGGAAGAAGGGCAGGGAG
ACGTGTCTGAAGGGCGTGGTTGTGTTCTCGGCAGGCTGGGATGTTTTTGCGTGCACAGCG
TCACTGGAGAAGCCTCTGGACCGCTTCCTCTTCAACTCTTACCTCACTAGCTGCCTGCAG
GCAGGTGTGGACAGGTGGGTGGGGGCCAGAGGAAGCTGGAGCCTGGGCTGGGGGATTAGC
CTACTTTAGCTGTTAGTTGCACTGAGCTACATTGTTTTCCTTTTAGCGGCTAGCACCTGA
CCCCTGAGGGTTGCGGAGGGTGTGAACTAGCTTGGCCACAAGCCTCTCTTTCTTTTTCTT
TCTTTCTTTCTCTCTTTCTTTCTTTCTTTCTTTCTTTCCCTCTCTATCAATCCCTCTCTC
TCGCTCTCACTCCCCTCCCTCTCTCTCGCTATCTCTCTCTGGGGATCCTGGACATATTTT
ATACTCTGTACTTTACTCTCAGAACCCAATTATATCATGAACATTGCCACTGGATATTAA
AGCAAACTCCCATGCTTCCCCATGATCCCCATTGAAGCAAACCTACTTCCTGCTTCCTCA
TGTTTTTCTCTCCTCTAGACACAAATCTGTCCTTGAGAAGAGTTACGACATTGATCATGT
GATGAAGGTAAAATGCACTTCCTTTTTAATGACCTCCTGTAAATCCCAATCAAAGTCAAA
GAAAGCAATCTATTCTATCCACGTTCTAGTCTCGTCCTTATGGCCCTCTTGGACTGCTGC
CAAATGTTGTTTATCCCAACCCCCATGCAGGCCAAGACCATCCGTGAGTTTGATGAGAGG
TTCACGTCTATAATGTTTGGCTACCCCACCAACGATGACTACTACCACGATGCCAGTCCC
ATCCACAAGCTCAAGTCTGTGCAGGTGCCCATGTTGTGTCTAAACGCTGCAGACGACGTC
TTCTCCCCCTGTCATGGTGAATCCACAATCACACCCCACCCCTGGCCCTCCATGTTGGTT
GCTTAGCCGATGAGCAAGGCCTATACGGAAAGACAATGCAGGAAAGACAATGAGTGTTGA
TATAGAACAGTGGTTCTCAAGCGTTTTGGGGTTGGGGACCCCTTTTGTGATAACAAATTC
ATCAGGGACCACCTTTAAATCAGAGCACAAGAAAATATCATACATGGAGTTCTGCTATGA
CTTTGGCAGTGCATGCCGGATGAACCACGTCTCGGGCCCTGGGAAGGCACATTTTGGCAC
ACGTTTTGCCATGAGGCAGA
>omy28:11608637-11610636
ACAGAGCTGTTCAACCGATCACAACCTATCTGGTTTTTACCACTGACACTCAATCAATCA
CTCAATGGCATGCAGCCATTGCCACTCAATCTGCGATATCCATTCAAATACCGTCATTAT
ACACAATGGCCTCTTAGGACTCCATACAACTAGGTCAATATTATTTTGGTTGAAGAGGGA
TAAGTGAGGAAAATGAATATCAGCATGCCGCTCTAGAATACGATGCATGACTTGTACAGG
AGAGAAGCTGTCATTGATAGACAAGAGGTGAAGTATTAATATGTGTATGGTCTCTAACCT
GTGGGGGCCCGCTGAGCAGCAGGCGTCGTGGGTGTGGCCCTGGGGGGGCCTCGTGGTCGG
CATGATGTTGCCTGGCTACGGTCCACTGTCTGTAGAAGAGGCTCTGCTCTGACCCTTGTA
AGTTAGGGGTGAAGAGGGGCCTCAGCGGGCTGCTCCAGCCCACATCGGCGTGAGGCAGCA
GGGAGAAGGAGCTCAGCTGGCTCCCTGATTCTCCAGCCAGGAGGTTGTAGGCAGCGCGCG
ACAGGATGACAGTCCGGGGAGAGACCCGAGGCGACTTGCTGCCCAGGCCGCATTGCGTAG
ACTGGCTGGGGTGCTGGGTCGAAGAGGAGGAGGAGGAAGAGGAGGAGGAGGAGCAGAAAG
AGGGAGGGGCGTTGGAAGGGCGGGAGGTGGCACTGCCCAGGGAGTCACACTCCTGTTTGA
CCCGGCGGGGGGATGTTGAGGTCGTGCCAGTCCCGTCTTCCATCATGATGCTGCCCGCCT
TGCCTCCAGGGGGGCAGGAGGAGGTGGAGGGCTTCTGGGAGGAGTCAGGGATGTCAGAGG
AGCTGACCCTGTTCTCTGCCAGAGAGCCTGAGGAAGGGAGAGGCATGGGGCGGTGAGAGA
GAGACTTCAGACTCATAGTACAGTATTTCAAAATGATTTCATAAACACAACTGTAACACC
ATCACTACGCAGTATTCTAAACCCTTACAACAATCATCTCGTCCCCATTTCCTAAACATG
TTGGGTCACTTTTGTCCAGCGACCTACCAGGGCCTCTGGTAGCAGCGCTGCTGTTGCTCT
GGGGCCTCTCCAGGTCCATGTGCAGCTCTAGGTCTGGGTCCTGAGGGCCACTGGTCTTGT
CTGTCTCTGGGTTCTTGGGGGCCATGGACCTGCAGGGCTCCTTTACCATCCTCAGGGACA
CGTAAGAGCAGGCCAGACCCACCTCCTGCTCCAGGGCCGTACGTGTCAGGTAGCTGGAGT
CGATCAGACGCAGGTCACAGTACTTCAGGGACCTAAGGGGGAAAGAACCAGTATAAAGGT
GAAGTGGAGGATTTTAATCGATTCTACAAAGTCAATCCAATTGACGATGCACCTTCTACG
GCAGATGTAGATGCTGTCATGTCAGTAGTACCTTGGAAAGGTTTCTCCTAGGGGGTCTTT
TCCAGTGAGGATGACAATGCAGGGAAGACCCTCCAGATCCTCATAGGTGTGAGGGACCCA
GTCTTCAATTTCACAGCCCCGCCAGGCCTGCCGAAACAACACAACACACAATTAGAAATG
TTTCGGAGCCAAATTAAAACCAGGTTGTTATGTCTCAGTCTGATGAAAGTCGTTGATATG
ATGACATTTTTTTAAAGACATCTTCATCAAATCACTTCTTTCATAGAGAGGGGTGGTGTT
CAAGGTCATCTCAATGTCAAATCATTTCTGGGTAACAATTAAGTACTTGTTTAATTGTGA
TTGTTTTTAATTAAAATGACCCCCCCCCCAAAAAAAAAAAAAAAAAAAAATTATATACAG
TGCCTTGCGAAAGTATTTGGCCCCCTTGAACTTTGCGACCTTTTGCCACATTTCAGGCTT
CAAACATAAAGATATAAAACTGTATTTTTTTGTGAAGAATCAACAACAAGTGGGACACAA
TCATGAAGTGGAACGACATTTATTGGATATTTCAAACTTTTTTAACAAATCAAAAACTGC
AAAATTGGGCGTGCAAAATT
>omy28:11856799-11858798
GACTAGTTGAGTATCTGGAGCATCAGTATTTGTGGGTTCGATTACAGGCTCAAAATGGCC
AGAAAAAATGTTTCTTTCTTCTGAAACTCATCAGTCTATTCTTGTTCTGAGAAATGAAGG
CAATTCCATGTGAGAAATTGCCAATAAACTGAAGATCTCGTACAACGCTGTGTACTACTC
CCTTCATAGTACAGGGCAGACTGTCTCTAACCAGAATAGAAAGAGGAGCGGGATGCCCCG
GTGCACAACTGAGCGAGAGGACAAGTACATTTGAGTGTCTAGTTTGAGAAACAGTCACAT
CGCAAGCCCACAACTGGGAGCTCTTAAGGATTGGTCCCTTTAAAAAAATGTTTTGGCCTA
AAATTATATACCCAAATCTAACTGCCTGTAGCTCAGCCCTGAAGCAAGGATATGCATATT
CTTGGTACTAAACACTTTGAAGTTTGTGGAAATGTGAAAGGAATGCAGGAGAATATAACA
TATTAGATCTGGTAAAAGATAATACAAAGAAAAAAACAACTGTTTTAAAAAAAAAAAATT
GTACCATCATGTTTTAAATGTAAGAGAAAGGCCTTAATGTATTATTCCAGTCCAGGTGCA
ATTTGTATTTAGCAGTGTACGTGCAAAGTTTTAGACTGATCCAATGAACCTTTGCATTTC
TGTTGAAAATGTTGTATCAAGACTGCCCAAATGTGCCTAATTTGTTTATTAATACCTTTT
CATTTTCAAATCTGTGCACTCTCCTCAAACAATAGCATGGTATTATTTCACTGTAATAGC
TACTGTAAATTGGACAGTGCAGTTAGATTAACAAGAATTTAAGCTTTCTGCCAATATCAG
ATATCAGACATTTTCTTGTTACTTACAACCTCATGCTAATCACATTAGCCTACGCTAGCT
CAACTGTCCTGCAGGGGACCCAATCCTGAAGACCCGCAAAACACCAGCTTCAATGTCAAC
AGTGAAGAGGCGGACTCCGGGATGCTGGCCTTCTAGGCATAGTTCCTCTATCCAGTGTCT
GTGTTCTTTTGCCCATCTTAATCTTTTCTTTTTATTGGCCAGTCTGAGATATGTCTTTTT
CTTTGCAACTCTGCCTAGATGGCCAGCATCCTGGATTGGCCTCTTCACTGTTAACGTTGA
GACTGGTGTTGTGTGGGTACTATTTAATGAAGATCTCATACAAGAATCTCAAATCTAAAA
TGTATTTTGATTTATCTAACACTCTTTTGTTAATACATGATTCCATATATGTTATTTTAT
AGTTTTGATGTCTTCACTATTATTATGCATTGTAGAACATAGTAAAACATTAAGAAAAAT
CCTTGAATGAGTAGGTGTTCTAAAACTAAATATATATATAAATATACAGATATACAGTGG
GGACTGAAATGATTGACACCCTTGATAAAGAGGAGCAAATATGACTGTATAAAATGTTAA
GGTCAACGGCATCATGAACTTTACCCAGTACCAGAACATTTGAGCAAAAAAACTGGTTCT
CTCTGCCAGGAGGTTGAAATTTGTCCGCATGTGGATGTTCGAGCAAGACAATAACACCAA
GCTCCTGAACATCTAATCAAATGGCTACCCAGACTATTTGCATNNNNNNNNNNCATAACC
TCAAAAGGATTTCCTCTGATGTTGGGAGGTTGCTCAGGTATTATAAAGACTGAATGGTAT
ACAGTTGGGTCCATTATTGGCACACTTAATAAAAGATGAGCAAAAACTTTTATACAAATA
TTGAGCTTTATTGTAAAAAAAAAAAGAAGAGGGAAATTATATTATTTTATACTAATAAAA
TTGTTGGGTTCTTTTATGCTTATGCATTCTCATTTCGACGCCAAACTCACAACTGGTGTG
TGCGTGGCCAAAGAGGTCTATCTTCACTGGTGTACGTGGCCAAAGAGGTCTATCTTCACT
GGTGTACGTGGCCAAAGAGGTCTATCTTCACTGGTGTNNNNNNNNNNNNNNNNNNNNNNN
NNNNNNNNNNNNNNNNNNNN
>omy28:12666405-12668404
GTGCGTGCGTGCGTGTGTGTGTGTGCGTGTGTGCGTGTGTGTGTGTGTGTGCGTGTGTGT
GTGTGTGCGTGCGTGTGTGCGCAAATTCGTATTTGTTTGTGTTTGTGAGTGTGTTTTTCC
CGAAGGGCGGGTGTAAGATATACGCCCTGTGTTATCTCAAGTTGTAACAGGCTGTCCACA
CATCTGAACTGTCACTCTTTAGACAGCTAGGAAGCGATGAGGAAGATCCCCAACATATGG
ATGCGACCGCTCTTATGGACCTGTATGACCGGACGATGGGAGGATGTCTGCAGGATCTCG
CCTACTGAAACCAGGTCGTGGTATGTCGAAAATGAGCATTTATAAACTTTTTTTTTTATC
TTACTTTCAGATGACTGTGTATATATGACAGGAATAACTTTTCTCCCACATACAGTTAAT
AGATTTTAAAATTAATAGATTTTACCCAAACAAAATAAAATAATTTAGCACTGCCTGTGG
CATTTTGGTATTTCTATTCATGATGCAGTGAGTAATATTGATAAACATTGTTGATACACA
TGTTTTGATTGAGCAATTGATTGAGGGATAATATTAGATTGAAAAATACATTTGAGAGAA
AGTTGTCATAAATTGAGAATGAGGCTCATGCAGAATGAGGCACCAGTGGTCCTTTGGAGC
TCAGTTGGTAGAGCATGGGGCTTGAAACGCCAGGGTAGTGGGATTGATTCCTGGGACCAC
CCATACATAAAATGTATGCGCCCATGACTGTGAATTGCTTTGGATAAAAGTGTCTGCTAA
ATTACATATATATTATAACATACATAATCCCCAATTTATATACTAGCTTTACAAAGACTA
TTTCAAGCATCGACTGCAGCTTGTTATTTTGTTATTTCAAAACCTGCAGGGAAATAGTGA
TTGCTCTGCTCTCACGTCCGTCCTTGTGGCATCTTGGCTTATATGTATATTTGACTGGAT
GCCATGCAGTTGTTTCCCCCTAACCCGACCTGTCAGACAACATCACCTGCCCCTGTGCTC
TGTCTTAATCACCGTGTCTGTAGATCAACTGACTGGGGATATACAGTGTATTCTGAAAAA
TATTCAGACCCCTTGACGTTTTCCTAATTTTGTTATGTTTCAGCCTTATTCTAAAATTGA
TTCAATAGTTTTTTCCCCTCATCAATCTACATACTGTCAAAGCATAATGTCAAAGCAAAA
GCAGGTTTTCAATTTTTGTATAATATCACATTTACATAAGTTTTCAGACGCTTTACTTAG
TACTTTGCTGAAGCACCTTTGGCAGCGATTACAGCCTTGAGTCTTCTTGGGTATGATGCT
ACAATACAAGCTTGGCCAACCTGTATTTGGGCAGTTTCTCCCATTCTTCTCTGCAGATCT
TCTCAACCTCTGTCAGGTTGGATGGGGAGCGTCGCTGCACAGCTATTTTCAGGTCTCTCC
AGAGATATTCGATCGGATTCAATTCCGGGCATTGGCTGGGCCACTCAAGGATATTCAGAG
ATTTGTCCTGAAGCCACTTCTGTGTTGTCTTGGTTGTGTGCTTATGGTCATTGTCCTATT
GGAAGGTGAACTTTTACCCCAGTCTGAGGTCCTGAGCGCTCTGGAGCAGGTTTTCATCAA
GGATCTCTCTGTACTTTGCCCCATTCATATTTTCCTTGATCCTGACTAGTCTCCCAGTCC
CTGCAGCTGCAAGATATTCCCACAGCATGATGCTGCCACCACCATGCTTCACTGTAGGGA
TGGTGCAAAGTTTCCTCCAGACGTGACGCTTGGCATTCAGGCCAAAGGAGTTCAATCTTG
GTTTCATCAGACCAGAGGATCTTGTTTCTCATGGTCAGAGTCCTTTAGGTGCCTTTTGGC
AAATTCCAAGAGGACTGTCGTGTGCCTTTTACTGAGGAGTGGCTTCCATCTGGCCACTCT
AGCATAAAGGCCTGATTGGTAGAGTGCTGCAGAGATGGTTTTCCTTCTGGAAGATTTTCC
AATCTCCACAGAGGAACTCT
```
And I have saved that into `/Users/eriq/Documents/git-repos/genoscape-bioinformatics/chinook-scripts/plink/chinook_top4.fasta`

